File: src/ClaudeToolbar.Core/Credentials/CredentialsPaths.cs
namespace ClaudeToolbar.Core.Credentials;

public static class CredentialsPaths
{
    public const string FileName = ".credentials.json";
    public const string ConfigDirVariable = "CLAUDE_CONFIG_DIR";
    public const string SettingsFileName = "settings.json";

    public static string Resolve(string? claudeConfigDir, string userProfile)
    {
        var dir = string.IsNullOrWhiteSpace(claudeConfigDir)
            ? System.IO.Path.Combine(userProfile, ".claude")
            : claudeConfigDir.Trim();
        return System.IO.Path.Combine(dir, FileName);
    }

    public static string ResolveFromEnvironment() =>
        Resolve(Environment.GetEnvironmentVariable(ConfigDirVariable),
                Environment.GetFolderPath(Environment.SpecialFolder.UserProfile));

    /// <summary>Claude Code's own settings file, next to the credentials file.</summary>
    public static string ClaudeSettingsPath(string? claudeConfigDir, string userProfile)
    {
        var dir = string.IsNullOrWhiteSpace(claudeConfigDir)
            ? Join(userProfile, ".claude")
            : claudeConfigDir.Trim();
        return Join(dir, SettingsFileName);
    }

    // Path.Combine always joins with the platform separator, so a forward-slash input on Windows
    // (e.g. "/Users/sam") would otherwise come back with a stray backslash mixed in.
    private static string Join(string dir, string leaf) =>
        dir.Contains('/')
            ? dir.TrimEnd('/', '\\') + "/" + leaf
            : System.IO.Path.Combine(dir, leaf);

    public static string ClaudeSettingsPathFromEnvironment() =>
        ClaudeSettingsPath(Environment.GetEnvironmentVariable(ConfigDirVariable),
                           Environment.GetFolderPath(Environment.SpecialFolder.UserProfile));
}

File: mac/Sources/ClaudeToolbarCore/Credentials/CredentialsPaths.swift
import Foundation

public enum CredentialsPaths {
    public static let fileName = ".credentials.json"
    public static let configDirVariable = "CLAUDE_CONFIG_DIR"

    public static func resolve(claudeConfigDir: String?, homeDirectory: String) -> String {
        let override = claudeConfigDir?.trimmingCharacters(in: .whitespacesAndNewlines) ?? ""
        let directory = override.isEmpty ? join(homeDirectory, ".claude") : override
        return join(directory, fileName)
    }

    public static func resolveFromEnvironment() -> String {
        resolve(claudeConfigDir: ProcessInfo.processInfo.environment[configDirVariable],
                homeDirectory: NSHomeDirectory())
    }

    public static let settingsFileName = "settings.json"

    /// Claude Code's own settings file, next to the credentials file.
    public static func claudeSettingsPath(claudeConfigDir: String?, homeDirectory: String) -> String {
        let override = claudeConfigDir?.trimmingCharacters(in: .whitespacesAndNewlines) ?? ""
        let directory = override.isEmpty ? join(homeDirectory, ".claude") : override
        return join(directory, settingsFileName)
    }

    public static func claudeSettingsPathFromEnvironment() -> String {
        claudeSettingsPath(claudeConfigDir: ProcessInfo.processInfo.environment[configDirVariable], homeDirectory: NSHomeDirectory())
    }

    private static func join(_ directory: String, _ component: String) -> String {
        if directory.hasSuffix("/") || directory.hasSuffix("\\") { return directory + component }
        return directory + "/" + component
    }
}
